### PR TITLE
feat(core/listener): listen for EventDataSignedBlock

### PR DESCRIPTION
**Pull request by walldiss**
## Overview

Resolves https://github.com/celestiaorg/celestia-node/issues/1640

### CODE REVIEW

File: core/fetcher.go [hunk 1]
```diff
@@ -78,6 +79,11 @@ func (f *BlockFetcher) GetBlockByHash(ctx context.Context, hash libhead.Hash) (*
 	return res.Block, nil
 }
 
+// GetSignedBlock queries Core for a `Block` at the given height.
+func (f *BlockFetcher) GetSignedBlock(ctx context.Context, height *int64) (*coretypes.ResultSignedBlock, error) {
```
Comment (renaynay) [inline core/fetcher.go:83]: why make this change in this PR? don't you only need it for core exchange PR?

**Review by renaynay — COMMENTED** [review]
otherwise lgtm

File: core/fetcher.go (hunk 1 above)
Comment (walldiss) [reply core/fetcher.go:83]: TestBlockFetcher_GetBlock_and_SubscribeNewBlockEvent is designed around checking results from sub against rpc call. I would either comment it out in this PR or combine both in one

Comment (renaynay) [reply core/fetcher.go:83]: ack

**Review by Wondertan — APPROVED** [review]

**Review by vgonkivs — APPROVED** [review]

**Pull request merged**